src/utils/src/time.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) 2023 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {AnimationConfig, Timeline, TimeRangeFilter} from '@kepler.gl/types';
+
+import {toArray} from './utils';
+
+export const TIMELINE_MODES = {
+  inner: 'inner',
+  outer: 'outer'
+};
+
+export const SAMPLE_TIMELINE = {
+  // value: [15], // represent 15% of the all width
+  value: [5, 15], // represent start at 5% and ends at 15%
+  domain: [1, 100], // represent the total domain
+  speed: 1,
+  enableInteraction: false, // can use interact with this timeline
+  isAnimating: false,
+  step: null,
+  // @todo: giuseppe coverType: 'inner' | 'outer'
+  mode: TIMELINE_MODES.inner
+  //  ....
+};
+
+export const getTimelineFromAnimationConfig = (animationConfig: AnimationConfig): Timeline => {
+  const {
+    currentTime,
+    domain,
+    speed,
+    isAnimating,
+    // @ts-expect-error not in anim
+    timeSteps,
+    defaultTimeFormat,
+    timeFormat,
+    timezone
+  } = animationConfig;
+
+  return {
+    // @ts-expect-error
+    value: toArray(currentTime),
```

Comment:
@macrigiuseppe is this correct? if the type of `Timeline` is `number | [number, number]`, shouldn't it be `value: currentTime` instead?

Reply:
given the current type this is correct. the Timeline structure is done to represents either a filter or an animation config. I do think we should enforce value to be an array of two elements because we expect to be as such in 

aninmation-controller.js
```
 _resetAnimationByDomain = () => {
      const {domain, value, animationWindow} = this.props;
      // interim solution while we fully migrate filter and layer controllers
      const setTimelineValue = this.props.updateAnimation || this.props.setTimelineValue;
      if (Array.isArray(value)) {
        if (animationWindow === ANIMATION_WINDOW.incremental) {
          setTimelineValue([value[0], value[0] + 1]);
        } else {
          setTimelineValue([domain[0], domain[0] + value[1] - value[0]]);
        }
      } else {
        setTimelineValue(domain[0]);
      }
    };
```